### DISCUSSION TITLE
Using Discovery Server with Fast-DDS-python

### DISCUSSION THREAD

**Discussion by MMarcus95** [eProsima/Fast-DDS-python#118] [Support]
Hello,

I would like to use Fast-DDS-python with Discovery Server. Is there a way to do that?

Thanks!

**Comment by elianalf** [comment]
Hi @MMarcus95, thanks for using Fast DDS.
Sure, you can use a Command Line Instruction to create a server, for more information refer to [CLI Discovery Section](https://fast-dds.docs.eprosima.com/en/latest/fastddscli/cli/cli.html#discovery) and you can simply use the environment variable `ROS_DISCOVERY_SERVER` to connect the server to a client, for more information refer to [ROS_DISCOVERY_SERVER Section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/env_vars/env_vars.html#ros-discovery-server).

**Comment by MMarcus95** [comment]
Thanks for the suggestion. Is there a way to do that using the python API instead? I know that with the C++ API you can set a domain participant to be the client of a server, specifying the remote server attributes and the discoveryProtocol as CLIENT.

For example, I have a server defined with C++ Fast-DDS API. Is there a way to define its client using python? Saying differently, is there a python version of [this](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html#id5) C++ code?

Thanks!!

**Reply by JesusPoderoso** [reply]
Hi @MMarcus95, there are no examples of that, but as long as the public API is exposed, the calls and arguments are the same in Python as they are in C++.
As @ploe24 pointed out, we are using SWIG too to wrap the C++ API into Python methods.

**Comment by ploe24** [comment]
> Is there a way to define its client using python? Saying differently, is there a python version of [this](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html#id5) C++ code?

You could write your client in C++, then wrap it using, for example SWIG, to create your Python API.
